src/sentry/buffer/base.py

```diff
@@ -34,21 +34,32 @@ class Buffer(Service):
 
     __all__ = ("incr", "process", "process_pending", "validate")
 
-    def incr(self, model, columns, filters, extra=None):
+    def incr(self, model, columns, filters, extra=None, signal_only=None):
         """
         >>> incr(Group, columns={'times_seen': 1}, filters={'pk': group.pk})
+        signal_only - added to indicate that `process` should only call the complete
+        signal handler with the updated model and skip creates/updates in the database. this
+        is useful in cases where we need to do additional processing before writing to the
+        database and opt to do it in a `buffer_incr_complete` receiver.
         """
         process_incr.apply_async(
-            kwargs={"model": model, "columns": columns, "filters": filters, "extra": extra}
+            kwargs={
+                "model": model,
+                "columns": columns,
+                "filters": filters,
+                "extra": extra,
+                "signal_only": signal_only,
+            }
         )
 
     def process_pending(self, partition=None):
         return []
 
-    def process(self, model, columns, filters, extra=None):
+    def process(self, model, columns, filters, extra=None, signal_only=None):
         from sentry.models import Group
         from sentry.event_manager import ScoreClause
 
+        created = False
         update_kwargs = dict((c, F(c) + v) for c, v in six.iteritems(columns))
```

Comment:
you can move the if not signal_only way up here to avoid all of this logic I believe